### PR TITLE
Make scio-test test suite compile faster

### PR DESCRIPTION
**Pull request by jto**
I've added some cache to speed up things a bit. 
I *think* that most of the compilation time is actually spent waiting because of classloading and locks in the compiler intermal when schemas are `eval`'d  in `ToMacro.safeImpl`. 

More specifically, there's a lot of waiting happening in [here](https://github.com/scala/scala/blob/2.13.x/src/reflect/scala/reflect/runtime/SymbolLoaders.scala#L135-L184).

Maybe we can avoid loading java classes by redesigning `Schema` ?

### CODE REVIEW

**Comment by jto** [comment]
There are some interesting comments regarding performances in Scalac source (https://github.com/scala/scala/blob/2.13.x/src/reflect/scala/reflect/runtime/Gil.scala#L19-L23)

**Comment by jto** [comment]
Closed it bc. it does not really improve compilation speed

**Comment by jto** [comment]
Reopening it bc. actually it does seem to make compilation a lot faster,

**Comment by jto** [comment]
aaaaannnnnd it's broken :/

**Comment by jto** [comment]
Fixed it :)

Fixed

File: scio-core/src/main/scala/com/spotify/scio/schemas/To.scala [hunk 1]
```diff
@@ -163,6 +164,51 @@ object To {
     macro ToMacro.safeImpl[I, O]
 }
 
+import scala.reflect.macros._
+import scala.tools.reflect.ToolBox
+import reflect.runtime.{universe => u}
+
+private[scio] final class FastEval(evalToolBox: ToolBox[u.type]) {
+
+  import scala.reflect.runtime.{universe => ru}
```
[the remark this answers is not in the document]
Comment (jto) [reply scio-core/src/main/scala/com/spotify/scio/schemas/To.scala]: Good catch

File: scio-core/src/main/scala/com/spotify/scio/schemas/To.scala [hunk 2]
```diff
@@ -163,6 +164,51 @@ object To {
...
+ */
+private[scio] object FastEval {
+  import scala.tools.reflect._
+
+  private lazy val tl: ToolBox[u.type] = {
+    val evalMirror = scala.reflect.runtime.currentMirror
+    evalMirror.mkToolBox()
+  }
+
+  def apply[T](ctx: blackbox.Context)(expr: ctx.Expr[T]): T =
```
[the remark this answers is not in the document]
Comment (jto) [reply scio-core/src/main/scala/com/spotify/scio/schemas/To.scala]: probably not. `FastEval` is private already

File: project/ScalacOptions.scala [hunk 3]
```diff
@@ -72,7 +72,9 @@ object Scalac {
       "-Xlint:constant", // Evaluation of a constant arithmetic expression results in an error.
       "-Ywarn-unused:imports", // Warn if an import selector is not referenced.
       "-Ywarn-extra-implicit", // Warn when more than one implicit parameter section is defined.,
-      "-Ydelambdafy:inline" // Set the strategy used for translating lambdas into JVM code to "inline"
+      "-Ydelambdafy:inline", // Set the strategy used for translating lambdas into JVM code to "inline"
+      "-Ybackend-parallelism",
```
[the remark this answers is not in the document]
Comment (jto) [reply project/ScalacOptions.scala]: It's just configuring the number of threads the compiler uses. Should not have negative impacts

[the remark this answers is not in the document]
Comment (regadas) [reply project/ScalacOptions.scala]: 👍

**Pull request merged**